aten/src/ATen/native/SegmentReduce.cpp

```diff
@@ -0,0 +1,81 @@
+#include <ATen/native/SegmentReduce.h>
+
+#include <ATen/ATen.h>
+
+#include <iostream>
+
+namespace at {
+namespace native {
+
+DEFINE_DISPATCH(segment_reduce_stub);
+
+enum ReductionType { MAX };
+const std::map<std::string, ReductionType> reduce2REDUCE = {
+    {"max", MAX},
+};
+
+Tensor _segment_reduce_cpu(
+    const Tensor& data,
+    std::string reduce,
+    const c10::optional<Tensor>& lengths,
+    const c10::optional<Tensor>& indices,
+    int64_t axis,
+    bool unsafe) {
+  TORCH_CHECK(axis == 0, "Currently only dim=0 is supported!");
+  TORCH_CHECK(data.dim() == 1);
+  TORCH_CHECK(!data.is_cuda());
+  TORCH_CHECK(data.numel() > 0);
```

Comment:
if possible, you should support empty tensors, by returning empty output tensor with the corresponding sizes.

Reply:
sg, will follow-up in future diff with test as well.